Modules/Core/Common/include/itkNeighborhoodOperator.h

```diff
@@ -87,19 +87,13 @@ class ITK_TEMPLATE_EXPORT NeighborhoodOperator : public Neighborhood<TPixel, VDi
   /** Slice iterator type alias support */
   using SliceIteratorType = SliceIterator<TPixel, Self>;
 
-  /** Sets the dimensional direction of a directional operator. */
+  /** Set/Get the direction (dimension number) of a directional operator. */
   void
   SetDirection(const unsigned long & direction)
   {
     m_Direction = direction;
   }
-
-  /** Returns the direction (dimension number) of a directional operator. */
-  unsigned long
-  GetDirection() const
-  {
-    return m_Direction;
-  }
+  itkGetConstMacro(Direction, unsigned long);
```

Comment:
This macro calls `this->Modified();`, which is declared in `itk::Object`. I guess that `itk::NeighborhoodOperator` does not derive from it.

Reply:
Yep, I did not dig further, but that was what I thought when I left the keyboard. Thanks Dženan.

Maybe I'm mistaken and there are good reasons not to do so, but I'd say that all ITK objects should be able to use the Get/Set macros. `itk::NeighborhoodOperator`'s base class does not inherit from `itk::Object`: 
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkNeighborhood.h

Maybe this was a justified design decision. Hopefully people will be able to comment.

Reply:
It may be a performance concern.  

itkFixedArray objects do not derive from itkObject for that reason.

Reply:
A `GetDirection()` member function defined by `itkGetConstMacro` would be _virtual_, which is probably not necessary. 

Makes me think, maybe we should have an `itkGetConstNonVirtualMacro` !